src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -281,17 +281,28 @@ private static IEnumerable<string> GetAllowedChildren(Type type, ErrorSink error
         /// </summary>
         internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink)
         {
-            return parentTag == null ||
-                TryValidateName(
+            if (parentTag != null)
```

Comment:
Instead of the extra nest, just do `if (parentTag == null) { return true; }`. Easier to read.